### PR TITLE
Added IAM API key support

### PR DESCRIPTION
**Pull request by stevenpkg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,176 +30,166 @@ understands that in both cases your intent is the same and responds accordingly.
...
+1. Once your app has been built and deployed, navagate to your IBM Cloud Dashboard and [import a workspace](#workspace).
 
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;![](readme_images/deploy-finished.png)
 
 ## <a name="usingCloudfoundry"></a> Using Cloudfoundry CLI tool to deploy your application
 
 To build the application:
 
-1 Download and install the [Cloudfoundry CLI](https://github.com/cloudfoundry/cli) tool.
+1. Download and install the [Cloudfoundry CLI](https://github.com/cloudfoundry/cli) tool.
```
Comment (germanattanasio) [inline README.md]: Check the README from https://github.com/watson-developer-cloud/assistant-simple/blob/master/README.md we don't use `cf` anymore. We use the IBM Cloud CLI now

File: package.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
     "gulp-useref": "^3.0.8",
     "helmet": "^3.1.0",
     "vcap_services": "^0.2.0",
-    "watson-developer-cloud": "^3.0.7",
+    "watson-developer-cloud": "^3.5.1",
```
Comment (germanattanasio) [inline package.json]: 3.7.0

File: routes/conversation.js [hunk 3]
```diff
@@ -14,17 +14,24 @@
...
-// Create the service wrapper
-const conversation = new watson.ConversationV1({
-  // If unspecified here, the ASSISTANT_USERNAME and ASSISTANT_PASSWORD env properties will be checked
-  // After that, the SDK will fall back to the bluemix-provided VCAP_SERVICES environment property
-  username: process.env.ASSISTANT_USERNAME || '<username>',
-  password: process.env.ASSISTANT_PASSWORD || '<password>',
-  url: 'https://gateway.watsonplatform.net/conversation/api/',
-  version_date: '2018-02-16'
-});
+var assistant;
```
Comment (germanattanasio) [inline routes/conversation.js]: If we instruct them to use IAM and the `.env` file then the constructor can be

```js
assistant = new AssistantV1({
  version: '2018-02-16',
});
```

**Review by germanattanasio — CHANGES_REQUESTED** [review]

File: routes/conversation.js [hunk 4]
```diff
@@ -20,16 +20,17 @@ var assistant;
 
 if (process.env.ASSISTANT_IAM_APIKEY !== undefined && process.env.ASSISTANT_IAM_APIKEY.length > 0) {
```
Comment (germanattanasio) [inline routes/conversation.js]: After doing this for all the demos, I realized that we could simplify this doing

```

const assistant = new AssistantV1({
  version: '2018-02-16',
});
```

The SDK is already looking at the Environment variables, so we don't need to do anything else. I tested it, and it works.

**Review by germanattanasio — CHANGES_REQUESTED** [review]

**Comment by stevenpkg** [comment]
German can you approve this?

**Review by germanattanasio — APPROVED** [review]

**Pull request merged**